testing/integration/upgrade_downgrade_test.go

```diff
@@ -41,25 +42,29 @@ func TestStandaloneDowngradeToSpecificSnapshotBuild(t *testing.T) {
 	ctx, cancel := testcontext.WithDeadline(t, context.Background(), time.Now().Add(10*time.Minute))
 	defer cancel()
 
-	// retrieve all the versions of agent from the artifact API
 	aac := tools.NewArtifactAPIClient()
 	latestSnapshotVersion, err := aac.GetLatestSnapshotVersion(ctx, t)
 	require.NoError(t, err)
 
-	// get all the builds of the snapshot version (need to pass x.y.z-SNAPSHOT format)
-	// 2 builds are required to be available for the test to work. This is because
-	// if we upgrade to the latest build there would be no difference then passing the version
-	// string without the buildid, being agent would pick that build anyway.
-	// We pick the build that is 2 builds back to upgrade to, to ensure that the buildid is
-	// working correctly and agent is not only picking the latest build.
-	builds, err := aac.GetBuildsForVersion(ctx, latestSnapshotVersion.VersionWithPrerelease())
+	// start at the build version as we want to test the retry
+	// logic that is in the build.
+	startFixture, err := define.NewFixture(t, define.Version())
+	require.NoError(t, err)
+	startVersion, err := startFixture.ExecVersion(ctx)
 	require.NoError(t, err)
-	if len(builds.Builds) < 2 {
-		t.Skipf("not enough SNAPSHOT builds exist for version %s. Found %d", latestSnapshotVersion.VersionWithPrerelease(), len(builds.Builds))
+
+	// when picking the build we need to find one that has a different commit hash to avoid file system
+	// collisions during the upgrade process.
+	// multiple builds can have different IDs but the same commit hash.
+	build, err := aac.GetLastBuildForVersion(ctx, latestSnapshotVersion.VersionWithPrerelease(), startVersion.Binary.Commit)
```

Comment:
To properly test the specific feature we should choose a build that is not the latest SNAPSHOT build.
If we pick the latest snapshot the result after the upgrade in the form `elastic-agent upgrade x.y.z-SNAPSHOT-<buildid>` would be indistinguishable from a regular `elastic-agent upgrade x.y.z-SNAPSHOT`

Reply:
@pchila can you elaborate more? 

* We get a list of snapshot builds of the same version as we're running
* Then we look for the latest snapshot that **does not match** the current hash of `main`
* This tests upgrading to the previous SNAPSHOT

Under what circumstances there will be a top/last snapshot with a different commit hash? My idea was that the commit hash filter will always find the previous snapshot that was not built with the same commit hash, is not what we want here?

Reply:
Changed in https://github.com/elastic/elastic-agent/pull/4095/commits/dd512b44169c539789719942ed60aa9fb3924c9f